ui-frontend/packages/catalog-ui-search/src/main/webapp/component/fields/useTimePrefs.tsx

```diff
@@ -0,0 +1,21 @@
+import { useState, useEffect } from 'react'
+import { useBackbone } from '../selection-checkbox/useBackbone.hook'
+// @ts-ignore ts-migrate(7016) FIXME: Could not find a declaration file for module '../s... Remove this comment to see the full error message
+import user from '../singletons/user-instance'
+
+const useTimePrefs = () => {
+  const { listenTo } = useBackbone()
+  const [, setForceRender] = useState(Math.random())
+
+  useEffect(() => {
+    listenTo(
+      user.getPreferences(),
+      'change:dateTimeFormat change:timeZone',
+      () => {
+        setForceRender(Math.random())
+      }
+    )
+  }, [])
```

Comment:
Do you need to stop listening to the model here?

Reply:
We don't need to, the stopListening is available if necessary, but the useBackbone effect cleans up the most common usecase which is unlistening on destruction of the component.